app/services/battle_service.rb

```diff
@@ -118,6 +121,78 @@ def ready_player(battle_id)
     broadcast_cable(battle_message)
   end
 
+  def unready_player(battle_id)
+    # Validate battle exists
+    battle = validate_battle(battle_id)
+
+    # Validate battle is readyable
+    validate_battle_readyable(battle)
+
+    # Validate the player is in the battle
+    validate_player_in_battle(battle)
+
+    unless battle.reload.players_ready?
+      BattlePlayer.find_by(battle: battle, player: @player).update!(ready: false)
+    end
+
+    battle.reload
+    # Broadcast battle update
+    message_hash = { type: PLAYER_READY, battle: battle }
+    battle_message = Entities::BattleMessage.represent message_hash, type: :include_players
+    broadcast_cable(battle_message)
+  end
+
+  def abandon_battle(battle_id)
+    # Validate battle exists
+    battle = validate_battle(battle_id)
+
+    # Validate battle is abandonable
+    validate_battle_abandonable(battle)
+
+    # Validate the player is in the battle
+    validate_player_in_battle(battle)
+
+    # set battle as abandoned for player
+    BattlePlayer.find_by(battle: battle, player: @player).update!(abandoned: true)
+
+    # If the battle has all players abandoned, move to abandoned state
+    if battle.reload.players_abandoned?
+      battle.abandoned!
+      type = BATTLE_ABANDONED
+    else
+      type = PLAYER_ABANDONED
+    end
+
+    # Broadcast battle update
+    message_hash = { type: type, battle: battle }
```

Comment:
Let's reload the battle in case the UI needs anything that's super up to date on the message. Same for others